src/Servers/Kestrel/Transport.MsQuic/Internal/QuicApi.cs

```diff
@@ -0,0 +1,200 @@
+
+        internal NativeMethods.RegistrationOpenDelegate RegistrationOpenDelegate { get; private set; }
+        internal NativeMethods.RegistrationCloseDelegate RegistrationCloseDelegate { get; private set; }
+
+        internal NativeMethods.SecConfigCreateDelegate SecConfigCreateDelegate { get; private set; }
+        internal NativeMethods.SecConfigCreateCompleteDelegate SecConfigCreateCompleteDelegate { get; private set; }
+        internal NativeMethods.SecConfigDeleteDelegate SecConfigDeleteDelegate { get; private set; }
+
+        internal NativeMethods.SessionOpenDelegate SessionOpenDelegate { get; private set; }
+        internal NativeMethods.SessionCloseDelegate SessionCloseDelegate { get; private set; }
+        internal NativeMethods.SessionShutdownDelegate SessionShutdownDelegate { get; private set; }
+
+        internal NativeMethods.ListenerOpenDelegate ListenerOpenDelegate { get; private set; }
+        internal NativeMethods.ListenerCloseDelegate ListenerCloseDelegate { get; private set; }
+        internal NativeMethods.ListenerStartDelegate ListenerStartDelegate { get; private set; }
+        internal NativeMethods.ListenerStopDelegate ListenerStopDelegate { get; private set; }
+
+        internal NativeMethods.ConnectionOpenDelegate ConnectionOpenDelegate { get; private set; }
+        internal NativeMethods.ConnectionCloseDelegate ConnectionCloseDelegate { get; private set; }
+        internal NativeMethods.ConnectionShutdownDelegate ConnectionShutdownDelegate { get; private set; }
+        internal NativeMethods.ConnectionStartDelegate ConnectionStartDelegate { get; private set; }
+
+        internal NativeMethods.StreamOpenDelegate StreamOpenDelegate { get; private set; }
+        internal NativeMethods.StreamCloseDelegate StreamCloseDelegate { get; private set; }
+        internal NativeMethods.StreamStartDelegate StreamStartDelegate { get; private set; }
+        internal NativeMethods.StreamShutdownDelegate StreamShutdownDelegate { get; private set; }
+        internal NativeMethods.StreamSendDelegate StreamSendDelegate { get; private set; }
+        internal NativeMethods.StreamReceiveCompleteDelegate StreamReceiveComplete { get; private set; }
+
+        internal NativeMethods.SetContextDelegate SetContextDelegate { get; private set; }
+        internal NativeMethods.GetContextDelegate GetContextDelegate { get; private set; }
+        internal NativeMethods.SetCallbackHandlerDelegate SetCallbackHandlerDelegate { get; private set; }
+
+        internal NativeMethods.SetParamDelegate SetParamDelegate { get; private set; }
+        internal NativeMethods.GetParamDelegate GetParamDelegate { get; private set; }
+
+        internal void RegistrationOpen(byte[] name)
+        {
+            QuicStatusException.ThrowIfFailed(RegistrationOpenDelegate(name, out var ctx));
+            RegistrationContext = ctx;
+        }
+
+        internal long Handle { get => (long)RegistrationContext; }
```

Comment:
Also, let's remove this too. Just have the QuicRegistration class track all this.